File: apps/redux-test/src/store/UIState.ts
export type Page = "PlansOverview" | "Plan";

export type OverviewPageState = {
  page: "PlansOverview";
};

export type PlanPageState = {
  page: "Plan";
  openPlanId: string;
};

export type UIState = OverviewPageState | PlanPageState;

File: apps/planet_plans/lib/state/ui_state.dart
abstract class UIState {
  final Page page;

  UIState(this.page);
}

class OverviewPageState implements UIState {
  final Page page = Page.plansOverview;

  OverviewPageState();
}

class SettingsPageState implements UIState {
  final Page page = Page.settings;

  SettingsPageState();
}

class PlanPageState implements UIState {
  final Page page = Page.plan;
  final String openPlanId;

  PlanPageState(this.openPlanId);
}

enum Page { plansOverview, plan, settings }
